tests/mixin-is-test.toit

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2023 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the tests/LICENSE file.
+
+import expect show *
+
+mixin MixA:
+  a-method: return 41
+
+mixin MixB extends MixA:
+  b-method: return 42
+
+mixin MixC:
+  c-method: return 43
+
+mixin MixD extends MixB with MixC:
+  d-method: return 44
+
+mixin MixE:
+  e-method: return 45
+
+
+class ClassA extends Object with MixD MixE:
+
+class ClassB extends Object with MixD:
+
+confuse x -> any: return x
```

Comment:
You may want a better confuse function here. This is a bit too easy for the type propagator.

Reply:
I think we should do that in a follow-up PR. For all `confuse` methods.